CMakeLists.txt

```diff
@@ -11,16 +11,31 @@ project(bpfilter
 
 include(GNUInstallDirs)
 
-find_package(Doxygen REQUIRED)
+option(BUILD_TESTING "Enable testing" OFF)
+option(BUILD_DOCS "Enable building documentation" OFF)
+
+if(BUILD_DOCS)
+    find_package(Doxygen REQUIRED)
+endif()
 find_package(PkgConfig REQUIRED)
 find_package(FLEX REQUIRED)
 find_package(BISON REQUIRED)
 
 pkg_check_modules(bpf REQUIRED IMPORTED_TARGET libbpf)
 pkg_check_modules(elf REQUIRED IMPORTED_TARGET libelf)
-pkg_check_modules(cmocka REQUIRED IMPORTED_TARGET cmocka)
+if(BUILD_TESTING)
+    pkg_check_modules(cmocka REQUIRED IMPORTED_TARGET cmocka)
+endif()
 pkg_check_modules(nl REQUIRED IMPORTED_TARGET libnl-3.0)
 
+find_library(ARGP_LIBRARY 
+    NAMES argp
+    PATHS ${CMAKE_SYSROOT}/usr/lib 
+    REQUIRED)
+if(NOT ARGP_LIBRARY)
+    message(FATAL_ERROR "libargp not found")
+endif()
```

Comment:
`ARGP_LIBRARY` is `REQUIRED` above, if not found, CMake will fail the generation and print an error message, no need for this condition.